### PR TITLE
lib: wifi_credentials: add support for setting MFP

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds management frame protection settings to the wifi_credentials library.

### CODE REVIEW

**Comment by maxd-nordic** [comment]
@krish2718 @bama-nordic @krga2022

File: subsys/net/lib/wifi_credentials/wifi_credentials_shell.c [hunk 1]
```diff
@@ -186,6 +194,17 @@ static int cmd_add_network(const struct shell *shell, size_t argc, char *argv[])
 		}
 	}
 
+	if (arg_idx < argc) {
+		/* look for mfp_disabled flag */
+		if (strcmp("mfp_disabled", argv[arg_idx]) == 0) {
```
Comment (gregersrygg) [inline subsys/net/lib/wifi_credentials/wifi_credentials_shell.c:199]:
```suggestion
		if (strncmp("mfp_disabled", argv[arg_idx], 12) == 0) {
```

File: subsys/net/lib/wifi_credentials/wifi_credentials_shell.c [hunk 2]
```diff
@@ -186,6 +194,17 @@ static int cmd_add_network(const struct shell *shell, size_t argc, char *argv[])
...
+			creds.header.flags |= WIFI_CREDENTIALS_FLAG_MFP_DISABLED;
+			++arg_idx;
+		} else if (strcmp("mfp_required", argv[arg_idx]) == 0) {
```
Comment (gregersrygg) [inline subsys/net/lib/wifi_credentials/wifi_credentials_shell.c:202]:
```suggestion
		} else if (strncmp("mfp_required", argv[arg_idx], 12) == 0) {
```

**Review by krish2718 — APPROVED** [review]

File: subsys/net/lib/wifi_credentials/wifi_credentials_shell.c (hunk 1 above)
Comment (balaji-nordic) [reply subsys/net/lib/wifi_credentials/wifi_credentials_shell.c:199]: Same suggestion, but use strlen('mfp_disabled') instead of 12.

**Pull request merged**